include/math/acos.h

```diff
@@ -0,0 +1,47 @@
+
+#pragma once
+
+#include "internal/defines.h"
+#include <type_traits>
+
+#if !defined(XTD_TARGET_CUDA) && !defined(XTD_TARGET_HIP) && !defined(XTD_TARGET_SYCL)
+#include <cmath>
+#endif
+
+namespace xtd {
+
+  template <typename T, typename = std::enable_if_t<std::is_floating_point_v<T>>>
+  XTD_DEVICE_FUNCTION
+  inline constexpr float acos(T arg) {
+#if defined(XTD_TARGET_CUDA)
+    // CUDA device code
+    return ::acos(arg);
+#elif defined(XTD_TARGET_HIP)
+    // HIP/ROCm device code
+    return ::acos(arg);
+#elif defined(XTD_TARGET_SYCL)
+    // SYCL device code
+    return sycl::acos(arg);
+#else
+    // standard C++ code
+    return std::acos(arg);
+#endif
+  }
+
+  XTD_DEVICE_FUNCTION
+  inline constexpr float acosf(float arg) {
```

Comment:
sorry This is utterly wrong! Did you check "assembly"? 
please call ::acosf at least for cuda and hip.
said that I wish to reminf that both cuda and hip support templated std::acos and produce correct code.
 
v.